x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/state_management.ts

```diff
@@ -67,11 +66,17 @@ export const getInitialState = (props: EditorFrameProps): EditorFrameState => {
   return {
     saving: false,
     title: i18n.translate('xpack.lens.chartTitle', { defaultMessage: 'New visualization' }),
-    datasource: {
-      state: null,
-      isLoading: Boolean(props.initialDatasourceId),
-      activeId: props.initialDatasourceId,
-    },
+    datasourceStates: props.doc
+      ? _.mapValues(props.doc.state.datasourceStates, state => ({ isLoading: true, state }))
+      : props.initialDatasourceId
+      ? {
+          [props.initialDatasourceId]: {
+            state: null,
+            isLoading: true,
+          },
+        }
+      : {},
```

Comment:
Not a fan of the double ternary style here, it's pretty confusing to read

Reply:
I agree, rewrote it.